streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java

```diff
@@ -64,13 +64,15 @@ public void onPartitionsAssigned(final Collection<TopicPartition> assignedPartit
         final long start = time.milliseconds();
         List<TopicPartition> revokedStandbyPartitions = null;
 
+        final int assignmentErrorCode = streamThread.getAssignmentErrorCode();
         try {
             if (streamThread.setState(State.PARTITIONS_ASSIGNED) == null) {
                 log.debug(
                     "Skipping task creation in rebalance because we are already in {} state.",
                     streamThread.state()
                 );
-            } else if (streamThread.getAssignmentErrorCode() != AssignorError.NONE.code()) {
```

Comment:
@mjsax I don't believe this is necessary to do the fix, but I think it's the right thing to do -- WDYT?

Reply:
I actually think we would need to `pause()` all partitions on version probing, too, to avoid that `poll()` returns any data to the `StreamThread` -- we don't check the VP flag directly after `poll()` and want `runOnce()` to just be a no-op for the VP case. \cc @guozhangwang

Reply:
We synced on this and confirmed this is not a problem, as we do pause the partitions in `taskManager#createTasks` -- however we're also pulling this part out into a separate PR

Reply:
System test to confirm this piece is not necessary for the PR: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3364/

Reply:
Sorry to go back and forth, but actually I believe this part is absolutely necessary to do along with the other fix in this PR -- also we can just take out the check entirely, as we already test for the only other `AssignorError` at the top of this callback